In [2]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
import seaborn as sns



1. It is assumed that the mean systolic blood pressure is μ = 120 mm Hg. In the Honolulu Heart Study, a sample of n = 100 people had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. Is the group significantly different (with respect to systolic blood pressure!) from the regular population?

    + Set up the hypothesis test.
    + Write down all the steps followed for setting up the test.
    + Calculate the test statistic by hand and also code it in Python. It should be 4.76190. What decision can       you make based on this calculated value?



In [8]:
mean=120
n=100 #
X=130.1
standard_dev=21.21
mean=120

In [9]:
#test statistic for a Z-test is the Z-statistic, which has the standard normal distribution under the null hypothesis
#z = ( sample_mean - pop_mean ) / ( pop_std_dev / np.sqrt(45) )
#z=X-mittelwert/standardabweichung

z = ( X - mean )/( standard_dev / np.sqrt(n))

In [13]:
print("test statistic:", z)

test statistic: 4.761904761904759


In [22]:
zc = stats.norm.ppf(0.95)
zc

#The critical value corresponding to a 0.95 area of a normal distribution is: 1.64

1.6448536269514722

In [28]:
zc = stats.norm.ppf(1-(0.05/2))
zc

1.959963984540054

What decision can you make based on this calculated value?

there must be a significant difference from the regular population because HO is rejected 

#z= 4.761904761904759 > 1.6444 -> reject H0 (if z<1.96 -> H0 not rejected)

# 2.

We will have another simple example on two sample one sided t test. In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file Data/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other

In [14]:

#2sided t-test:


#H0: machine same
#H1: new machine faster
#n10


#hypothesis: new machine will pack faster -> #HO 

In [5]:
data=pd.read_csv("./Data_Machine.txt", sep="\t")

In [6]:
data

New Machine    Old Machine
0         42.1           42.7
1         41.0           43.6
2         41.3           43.8
3         41.8           43.3
4         42.4           42.5
5         42.8           43.5
6         43.2           43.1
7         42.3           41.7
8         41.8           44.0
9         42.7           44.1

In [12]:
data.columns = ["new_machine", "new_machine"]

In [21]:
#1.mean



mean_newmach=np.mean(data["new_machine"])
mean_oldmach=np.mean(data["old_machine"])

print("Sample mean of the new machine is:" , mean_newmach)
print("Sample mean of the old machine is:" , mean_oldmach)

Sample mean of the new machine is: 42.14
Sample mean of the old machine is: 43.230000000000004


In [26]:
#2.std

std_newmach=np.std(data["new_machine"])
std_oldmach=np.std(data["old_machine"])
n_newmach=10
n_oldmach=10


print("The sample standard deviation of the new machine is:" , std_newmach)
print("The sample standard deviation of the old machine is:" , std_oldmach)

The sample standard deviation of the new machine is: 0.6483826030978941
The sample standard deviation of the old machine is: 0.7114070564732956


In [35]:
t = ( mean_newmach - mean_oldmach ) / np.sqrt( ((std_newmach**2)/9) + ((std_oldmach**2)/9 ))
print("t value: ", t)
#we need to use the t-Student distribution. The number of degrees of freedom is n-1 from the smallest sample size

t value:  -3.3972307061176026


In [33]:
#df = sample_size_1 + sample_size_2 - 2 = 18
tc = stats.t.ppf(1-(0.05/2),df = 18)
print("t critcal value: ", tc)
tc

t critcal value:  2.10092204024096


2.10092204024096

In [ ]:
#tc>t - > recejt H0-> machine is faster